meson.build

```
@@ -0,0 +1,64 @@
+project(
+  'NumPy',
+  'c', 'cpp', 'cython',
+  # Note that the git commit hash cannot be added dynamically here
+  # It is dynamically added upon import by versioneer
+  # See `numpy/__init__.py`
+  version: '1.24.0.dev0',
+  license: 'BSD-3',
+  meson_version: '>= 0.64.0',
+  default_options: [
+    'buildtype=debugoptimized',
+    'c_std=c99',
+    'cpp_std=c++11',
+    'blas=openblas',
+    'lapack=openblas',
+    'pkgconfig.relocatable=true',
+  ],
+)
+
+fs = import('fs')
+
+cc = meson.get_compiler('c')
+cpp = meson.get_compiler('cpp')
+
+# Check compiler is recent enough (see the SciPy Toolchain Roadmap for details)
+if cc.get_id() == 'gcc'
+  if not cc.version().version_compare('>=8.4')
+    error('NumPy requires GCC >= 8.4')
+  endif
+elif cc.get_id() == 'msvc'
+  if not cc.version().version_compare('>=1910')
```

```suggestion
  if not cc.version().version_compare('>=19.10')
```

Needed for windows build

---

Thanks Melissa.

That means vc141, I'm wondering if that's still the correct lower bound. We build wheels with vc142 now, and conda-forge has moved as well. But I think vc141 still works, and we may want to allow others to build with it. On the other hand, SciPy already uses vc142 as the lower boun.d @charris thoughts?

---

The main reason we used 1.41 was anaconda compatibility. I favor using the latest version that doesn't cause major problems with the community. Sounds like 1.42 would be fine these days.

We will need to move on from vs2019 eventually, we should think about how to manage that so that it goes smoothly. Maybe propose some dates?

---

Okay, `vc142` it is.

> We will need to move on from vs2019 eventually, we should think about how to manage that so that it goes smoothly. Maybe propose some dates?

The issue I think is mainly the `libnpymath` and `libnpyrandom` static libraries. A search turns up very few users, only SciPy so far. I'd say we deprecate those, and then upgrading should be much easier in the future.

---

I'll note that that static library issue is mostly gh-20880.